src/components/WineLister.jsx

```diff
@@ -1,16 +1,30 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { useWineList } from '../contexts/WineListContext';
 import WineBottle from './WineBottle';
 
-function WineLister() {
+function WineLister({ text }) {
   const { wineList } = useWineList();
+  console.log(wineList);
+  console.log(typeof wineList);
+  console.log(text);
+  console.log(typeof text);
   return (
     <>
-      { wineList && wineList.map((wine) => (
-        <WineBottle key={wine.id} wine={wine} />
-      ))}
+      { wineList && wineList.filter((wine) => (wine.estate.toUpperCase()
+        .includes(text.toUpperCase())) || (wine.appellation.toUpperCase()
+        .includes(text.toUpperCase())) || (wine.color.toUpperCase()
+        .includes(text.toUpperCase())) || (wine.year.toString(10)
+        .includes(text)))
+        .map((wine) => (
+          <WineBottle key={wine.id} wine={wine} />
+        ))}
```

Comment:
excellent :)

if you want to test each property, you can try `Object.values()` to get an array with all the values in an object:

    wineList.filter((wine) => (
      (Object.values(wine).join().toUpperCase().includes(text.toUpperCase())
    ))